nrf54L dt fixes

SebastianBoe opened the pull request:
misc. nrf54L DT fixes

SebastianBoe committed:
boards: nordic: 54l: use wdt31 instead of wdt30

Use wdt31 instead of wdt30 as wdt30 is hardware fixed to secure.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe committed:
dts: nordic: 54l: Don't define wdt30 for the non-secure domain

Don't define wdt30 for the non-secure domain as it is hardware fixed
to secure.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe committed:
dts: nordic: 54l: Don't define UICR for the non-secure domain

Don't define UICR for the non-secure domain as it is hardware fixed to
secure.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe committed:
dts: nordic: 54l: Change the peripheral address map for ns

Define peripherals with the 0x4000_0000 address range when building
for non-secure.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe committed:
boards: nordic: 54L: Refactor the board DT

Refactor the 54L board file to be able to better support out-of-tree
non-secure boards.

This aligns better with nrf53's DT.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

rlubos commented on boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15-common.dtsi:
```diff
@@ -60,7 +60,7 @@
 		sw2 = &button2;
 		sw3 = &button3;
-		watchdog0 = &wdt30;
+		watchdog0 = &wdt31;
```
As we change the default watchdog instance here, we need to align the watchdog sample with this change, specifically those files:
https://github.com/zephyrproject-rtos/zephyr/blob/531c457550ca4792b82f21dc1ceb3aaee23bde60/samples/drivers/watchdog/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay
https://github.com/zephyrproject-rtos/zephyr/blob/531c457550ca4792b82f21dc1ceb3aaee23bde60/samples/drivers/watchdog/boards/nrf54l15pdk_nrf54l15_cpuflpr.overlay
https://github.com/zephyrproject-rtos/zephyr/blob/531c457550ca4792b82f21dc1ceb3aaee23bde60/samples/drivers/watchdog/boards/nrf54l15pdk_nrf54l15_cpuflpr_xip.overlay
And test:
https://github.com/zephyrproject-rtos/zephyr/blob/531c457550ca4792b82f21dc1ceb3aaee23bde60/tests/drivers/watchdog/wdt_basic_api/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay

SebastianBoe replied:
done

anangl commented on dts/common/nordic/nrf54l15.dtsi:
```diff
@@ -104,11 +104,14 @@
+#ifdef USE_NON_SECURE_ADDRESS_MAP
+/* intentionally empty because UICR is hardware fixed to Secure */
+#else
+uicr: uicr@ffd000 {
```
The indentation could be improved here.

SebastianBoe replied:
done

SebastianBoe committed:
samples: wdt_basic_api: nrf: Add missing nrf overlays

Add missing nrf overlays to fix the sample at build time.

Copied from samples/drivers/watchdog/boards.

It is not clear why the build failure was not detected earlier.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

anangl reviewed (approved):
nit: the second commit title should start with "tests:", not "samples:".
